utils.js

```diff
@@ -236,6 +237,50 @@ class InstallerUtils {
     return data.toString(encoding).trim();
   }
 
+  /**
+   * Fetch a version number from a URL.  If not found, downgrade the major
+   * version and try again.  If a WebDriver release lags the browser release
+   * (which seems common), this will compensate.  Both Chrome and Edge use
+   * this.
+   *
+   * @param {number} idealMajorVersion
+   * @param {number} minMajorVersion
+   * @param {function(number): string} urlFormatter
+   * @param {string=} encoding
+   * @return {!Promise<string>}
+   */
+  static async fetchVersionUrlWithAutomaticDowngrade(
+      idealMajorVersion, minMajorVersion, urlFormatter, encoding) {
+    let majorVersion = idealMajorVersion;
+    let firstError = null;
+
+    while (majorVersion >= minMajorVersion) {
+      const versionUrl = urlFormatter(majorVersion);
+      try {
+        return await InstallerUtils.fetchVersionUrl(versionUrl, encoding);
+      } catch (error) {
+        if (error.cause.status != 404) {
+          // Any unexpected error (other than HTTP 404) is thrown immediately.
+          throw error;
+        }
+
+        // Save the first error in case we run out this loop.  We'll throw this
+        // one if none of the allowed versions can be found.
+        if (firstError == null) {
```

Comment:
nit; Can use a nullish coalescing assignment:

`firstError ??= error;`

Reply:
This code doesn't get compiled, and nullish coalescing operators are only supported in node.js >= 14.  That's not super new, but using that would also set a minimum requirement on this project and all dependent projects (like Shaka Player), so I would prefer not to use it.

Reply:
In particular, node.js minimum requirements are a pain for people who use a package manager for node.js installation.  Although node.js 14 came out in 2020, there is not yet an Ubuntu LTS release with node.js >= 14 in a native package.  (Ubuntu LTS from 2020 has node.js v10 from 2018, Ubuntu LTS from 2022 has node.js v12 from 2019, etc.  It sucks.)